src/js/components/HealthChecksComponent.jsx

```diff
@@ -184,10 +194,39 @@ var HealthChecksComponent = React.createClass({
                 }
                 fieldId={`${fieldsetId}.${i}.portIndex`}
                 label="Port Index"
-                value={row.portIndex}>
+                value={row.portIndex || 0}>
                 <input ref={`portIndex${i}`} {...numberInputAttributes} />
               </FormGroupComponent>
             </div>
+            <div className={portClassSet}>
+              <FormGroupComponent
+                errorMessage={
+                  getErrorMessage(`${fieldsetId}.${i}.port`)
+                }
+                fieldId={`${fieldsetId}.${i}.port`}
+                label="Port Number"
+                value={row.port || 0}>
+                <input ref={`port${i}`} {...numberInputAttributes} />
+              </FormGroupComponent>
+            </div>
+            <div className="col-sm-3">
+              <FormGroupComponent
+                errorMessage={
+                  getErrorMessage(`${fieldsetId}.${i}.portType`)
+                }
+                fieldId={`${fieldsetId}.${i}.portType`}
+                label="Port Type"
+                value={row.portType}>
+                <select defaultValue={row.portType} ref={`portType${i}`}>
+                  <option value={HealthCheckPortTypes.PORT_INDEX}>
+                    Port Index
+                  </option>
+                  <option value={HealthCheckPortTypes.PORT_NUMBER}>
```

Comment:
Why? Isn't directly related to the data. But OK we could call it ```portNumberClassSet```, doesn't hurtz.